Lib/test/test_threaded_pickle.py

```diff
@@ -0,0 +1,53 @@
+import sys
+import importlib
+import os
+import shutil
+import threading
+import unittest
+from test.support import TESTFN, forget
+import pickle
+
+slow_import_module = """
+import time
+
+time.sleep(%(delay)s)
+
+class ToBePickled(object):
+    pass
+"""
+
+
+class ThreadedPickleTests(unittest.TestCase):
+    def test_pickle_thread_imports(self):
+        # https://bugs.python.org/issue34572
+        delay = 0.01
+        os.mkdir(TESTFN)
+        self.addCleanup(shutil.rmtree, TESTFN)
+        sys.path.insert(0, TESTFN)
+        self.addCleanup(sys.path.remove, TESTFN)
+        contents = slow_import_module % {'delay': delay}
+        with open(os.path.join(TESTFN, "slowimport.py"), "wb") as f:
+            f.write(contents.encode('utf-8'))
+        self.addCleanup(forget, "slowimport")
+
+        import slowimport
+        obj = slowimport.ToBePickled()
+        pickle_bytes = pickle.dumps(obj)
+        del obj
+
+        # Then try to unpickle two of these simultaneously
+        def do_unpickle():
+            results.append(pickle.loads(pickle_bytes))
+
+        for repeat_test in range(10):
+            results = []
+            forget("slowimport")
+            self.assertNotIn("slowimport", sys.modules)
+            importlib.invalidate_caches()
```

Comment:
Can this affect any of the importlib tests?

Reply:
Any advice on how I can address this?  FWIW I pretty much copied this test from `test_threaded_import.py`.  

I haven't tested without that line, so it may not be necessary.  I'll see if the test works (i.e. fails without the patch) reliably without it.

Reply:
Actually, I don't know why it would affect the `importlib` tests (I'm the one who added that function IIRC).